### PR TITLE
fix: remove the size of the Icon compoent in v2

### LINKED ISSUE #2510: Remove the `size` of the `Icon` component in v2
### Description

Remove `size` from the `Icon` component.

Create the branch you are working on from `v2.0`.

### Problem Statement/Justification

_No response_

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by kaehehehe**
Closes #2510

## Description

I've removed the size of the Icon component in v2.0.

### CODE REVIEW

File: packages/components/icon/src/icon.tsx [hunk 1]
```diff
@@ -14,27 +9,17 @@ import { useToken } from "@yamada-ui/use-token"
...
+  UIProps
 
 /**
  * `Icon` is a general icon component that can be used in your projects.
  *
  * @see Docs https://yamada-ui.com/components/media-and-icons/icon
  */
 export const Icon = forwardRef<IconProps, "svg">((props, ref) => {
-  const [styles, { size, ...mergedProps }] = useComponentStyle("Icon", props)
+  const [styles, { ...mergedProps }] = useComponentStyle("Icon", props)
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:22]: Please change to

```diff
- const [styles, { ...mergedProps }] = useComponentStyle("Icon", props)
+ const [styles, mergedProps] = useComponentStyle("Icon", props)
```

File: .changeset/mighty-geese-cheer.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/icon": patch
```
Comment (hirotomoyamada) [inline .changeset/mighty-geese-cheer.md:2]: Please change `patch` to `major`.

File: packages/components/icon/src/icon.tsx (hunk 1 above)
Comment (kaehehehe) [reply packages/components/icon/src/icon.tsx]: I fixed it.

[7cc7194](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/7cc719424838901674353bac357fe96446e03804)

File: .changeset/mighty-geese-cheer.md (hunk 2 above)
Comment (kaehehehe) [reply .changeset/mighty-geese-cheer.md]: I changed it. [cb8ef85](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/cb8ef85aa5fda25cf1edebab9ce196ecb6b8216d)

**Review by hirotomoyamada — DISMISSED** [review]

**Comment by hirotomoyamada** [comment]
Thanks!

**Comment by hirotomoyamada** [comment]
@kaehehehe 

Hi, thanks for the fix.

Test seems to be failing, please check.

**Comment by kaehehehe** [comment]
I removed the size prop from all the places where the Icon component is being used.  [0a1d3f7](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/0a1d3f7357a73da090d98c95a669cad1429da3c5)

**Comment by hirotomoyamada** [comment]
@kaehehehe 

Hi, thanks for the fix.

There seems to be some duplication of work in the `main` branch.

Please wait while we make some adjustments.

**Comment by hirotomoyamada** [comment]
@kaehehehe 

Work has been completed on `v1`.You can work on it.

**Comment by hirotomoyamada** [comment]
@kaehehehe

Hi, Thanks for the PR!

A week has passed since this PR was created, so maintainers will be joining this PR.

**Comment by kaehehehe** [comment]
I've resolved the conflict; please take a look. @hirotomoyamada

File: stories/components/media-and-icons/font-awesome.stories.tsx
Comment (hirotomoyamada) [inline stories/components/media-and-icons/font-awesome.stories.tsx:1]: Do not change this file.

File: docs/contents/components/media-and-icons/icon/index.ja.mdx
Comment (hirotomoyamada) [inline docs/contents/components/media-and-icons/icon/index.ja.mdx:1]: Do not change this file.

File: docs/contents/components/media-and-icons/icon/index.mdx
Comment (hirotomoyamada) [inline docs/contents/components/media-and-icons/icon/index.mdx:1]: Do not change this file.

File: docs/pages/icons/icon-drawer.tsx
Comment (hirotomoyamada) [inline docs/pages/icons/icon-drawer.tsx:1]: Do not change this file.

File: docs/pages/icons/list.tsx
Comment (hirotomoyamada) [inline docs/pages/icons/list.tsx:1]: Do not change this file.

File: docs/pages/icons/not-found.tsx
Comment (hirotomoyamada) [inline docs/pages/icons/not-found.tsx:1]: Do not change this file.

File: stories/components/media-and-icons/lucide.stories.tsx
Comment (hirotomoyamada) [inline stories/components/media-and-icons/lucide.stories.tsx:1]: Do not change this file.

File: packages/components/menu/tests/menu.test.tsx
Comment (hirotomoyamada) [inline packages/components/menu/tests/menu.test.tsx:1]: Do not change this file.

File: packages/components/fontawesome/src/icon.tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export type IconProps = UIIconProps & IconOptions
  * @see Docs https://yamada-ui.com/components/media-and-icons/fontawesome
  */
 export const Icon = forwardRef<IconProps, "svg">(
-  ({ className, icon, size: fontSize, __css, ...rest }, ref) => {
+  ({ className, icon, fontSize, __css, ...rest }, ref) => {
```
Comment (hirotomoyamada) [inline packages/components/fontawesome/src/icon.tsx:30]: Please change

```
({ className, icon, fontSize, __css, ...rest }, ref) => {
```

to

```
({ className, icon, __css, ...rest }, ref) => {
```

File: packages/components/icon/src/icon.tsx [hunk 4]
```diff
@@ -14,27 +9,18 @@ import { useToken } from "@yamada-ui/use-token"
...
-  /**
-   * The CSS `font-size` property.
-   *
-   * @deprecated Use `fontSize` instead.
-   */
-  size?: CSSUIProps["fontSize"]
-}
-
 export type IconProps = Omit<SVGAttributes<SVGElement>, keyof UIProps> &
   Omit<ThemeProps<"Icon">, "size"> &
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:13]: Please change

```
Omit<ThemeProps<"Icon">, "size">
```

to

```
ThemeProps<"Icon">
```

File: packages/components/icon/src/icon.tsx [hunk 5]
```diff
@@ -14,27 +9,18 @@ import { useToken } from "@yamada-ui/use-token"
...
 /**
  * `Icon` is a general icon component that can be used in your projects.
  *
  * @see Docs https://yamada-ui.com/components/media-and-icons/icon
  */
 export const Icon = forwardRef<IconProps, "svg">((props, ref) => {
-  const [styles, { size, ...mergedProps }] = useComponentStyle("Icon", props)
+  const [styles, mergedProps] = useComponentStyle("Icon", props)
+
   let {
```
Comment (hirotomoyamada) [inline packages/components/icon/src/icon.tsx:24]: Please change `let` to `const`.

File: packages/components/loading/src/loading.tsx [hunk 6]
```diff
@@ -33,6 +33,8 @@ type LoadingOptions = {
    * The CSS `dur` property.
    */
   duration?: IconProps["dur"]
+
+  size?: CSSUIProps["fontSize"]
```
Comment (hirotomoyamada) [inline packages/components/loading/src/loading.tsx:37]: Please remove line.

File: packages/components/loading/src/loading.tsx
Comment (hirotomoyamada) [inline packages/components/loading/src/loading.tsx:1]: Please remove `size = "1em",` line.

**Review by hirotomoyamada — COMMENTED** [review]
Please change file `packages/theme/src/components/loading.ts`.

```
import type { ComponentStyle } from "@yamada-ui/core"

export const Loading: ComponentStyle<"Loading"> = {
  baseStyle: {},

  defaultProps: {
    variant: "oval",
    colorScheme: "primary",
    size: "1em",
  },
}
```

to

```
import type { ComponentStyle } from "@yamada-ui/core"

export const Loading: ComponentStyle<"Loading"> = {
  baseStyle: {},

  defaultProps: {
    variant: "oval",
    colorScheme: "primary",
    fontSize: "1em",
  },
}
```

File: packages/components/loading/src/loading.tsx
Comment (hirotomoyamada) [inline packages/components/loading/src/loading.tsx:1]: Change `Pick<ThemeProps<"Loading">, "colorScheme">` to `Pick<ThemeProps<"Loading">, "size", "colorScheme">`.

File: packages/components/fontawesome/src/icon.tsx (hunk 3 above)
Comment (kaehehehe) [reply packages/components/fontawesome/src/icon.tsx]: [7b4c209](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/7b4c209fdfbe99dc503db414e1b96089ee83106c)

File: packages/components/icon/src/icon.tsx (hunk 5 above)
Comment (kaehehehe) [reply packages/components/icon/src/icon.tsx]: [c1449a8](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/c1449a8f52780812c1522f8d85de494a1a9c1860)

File: packages/components/loading/src/loading.tsx (hunk 6 above)
Comment (kaehehehe) [reply packages/components/loading/src/loading.tsx]: [5b5b6ae](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/5b5b6ae65540fae1c82deda827822de21ece2694)

File: packages/components/loading/src/loading.tsx
Comment (kaehehehe) [reply packages/components/loading/src/loading.tsx:1]: [5b5b6ae](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/5b5b6ae65540fae1c82deda827822de21ece2694)

Comment (kaehehehe) [reply packages/components/loading/src/loading.tsx:1]: [5b5b6ae](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/5b5b6ae65540fae1c82deda827822de21ece2694)

File: stories/components/media-and-icons/font-awesome.stories.tsx
Comment (kaehehehe) [reply stories/components/media-and-icons/font-awesome.stories.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: docs/contents/components/media-and-icons/icon/index.mdx
Comment (kaehehehe) [reply docs/contents/components/media-and-icons/icon/index.mdx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: docs/pages/icons/icon-drawer.tsx
Comment (kaehehehe) [reply docs/pages/icons/icon-drawer.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: docs/pages/icons/list.tsx
Comment (kaehehehe) [reply docs/pages/icons/list.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: docs/pages/icons/not-found.tsx
Comment (kaehehehe) [reply docs/pages/icons/not-found.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: stories/components/media-and-icons/lucide.stories.tsx
Comment (kaehehehe) [reply stories/components/media-and-icons/lucide.stories.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: packages/components/menu/tests/menu.test.tsx
Comment (kaehehehe) [reply packages/components/menu/tests/menu.test.tsx:1]: [f1ff5c3](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/f1ff5c3738657483783c0f734d27a68a640b199d)

File: docs/contents/components/media-and-icons/icon/index.mdx [hunk 7]
```diff
@@ -88,3 +88,4 @@ import { faPoo } from "@fortawesome/free-solid-svg-icons"
 :::note
 If you want to learn more about `@yamada-ui/fontawesome`, please check [here](/components/media-and-icons/fontawesome).
 :::
+import { fontSizes } from "../../../../../packages/theme/src/tokens/font-sizes"
```
Comment (hirotomoyamada) [inline docs/contents/components/media-and-icons/icon/index.mdx:91]: Please remove line.

**Comment by hirotomoyamada** [comment]
@kaehehehe 

Hi, thanks for the fix!

https://github.com/yamada-ui/yamada-ui/pull/2537#discussion_r1729402907
https://github.com/yamada-ui/yamada-ui/pull/2537#pullrequestreview-2257944352

These two tasks are also requested.

File: docs/contents/components/media-and-icons/icon/index.mdx (hunk 7 above)
Comment (kaehehehe) [reply docs/contents/components/media-and-icons/icon/index.mdx]: [52160fa](https://github.com/yamada-ui/yamada-ui/pull/2537/commits/52160fa4bf4e01c0c95dd2115fe728e1092c377d)

**Comment by hirotomoyamada** [comment]
Thanks!

**Comment by quantumshiro** [comment]
it seems to be another conflict. plz fix it

**Pull request merged**